packages/editor/src/store/selectors.js

```diff
@@ -852,8 +852,8 @@ export function hasSelectedBlock( state ) {
  * @return {?string} Selected block client ID.
  */
 export function getSelectedBlockClientId( state ) {
-	const { start, end } = state.blockSelection;
-	return start === end && start ? start : null;
+	const selectedBlock = getSelectedBlock( state );
```

Comment:
We should not use `getSelectedBlock` in `getSelectedBlockClientId` as `getSelectedBlock` is not performant. I'd do the opposite. use ` getSelectedBlockClientId` inside `getSelectedBlock`

Reply:
The opposite is how it worked before, it was buggy :)

This fix in the selector tries to cover that the reducer doesn't take into account the `UNDO` / `REDO` actions. Explored a few ways to do 
 that but hit some deadends: https://github.com/WordPress/gutenberg/issues/12002#issuecomment-440752932

Reply:
Here's a proposal https://github.com/WordPress/gutenberg/pull/12214#issuecomment-441022823

We shouldn't sacrifice performance here. I personally think we should remove `getSelectedBlock` entirely if possible. :)

Reply:
Performance wise this can be improved, indeed. Going to update.

Reply:
0d538fc